File: arrays/moveElementToEnd/moveElementToEnd.js
function moveElementToEnd(array, toMove) {
  let moveOps = 0
  for(let i = 0; i + moveOps < array.length; i++) {
    
    if(array[i] === toMove) {
      let numToEnd = array.splice(i, 1);
      i--;
      moveOps++;
      array.push(toMove)
    }
  }
  return array
}

console.log(moveElementToEnd([2, 1, 2, 3, 4, 2], 2)) // [1, 3, 4, 2, 2, 2]
// Time Complexity: O(n²)
// Space Complexity: O(1)   

File: arrays/moveElementToEnd/moveElementToEnd.py
# O(n) time | O(1) space
def moveElementToEnd(array, toMove):
    i = 0
    j = len(array) - 1
    while i < j:
        while i < j and array[j] == toMove:
            j -= 1
        if array[i] == toMove:
            array[i], array[j] = array[j], array[i]
        i += 1
    return array


print(moveElementToEnd([1, 2, 3, 4, 5, 6, 7, 8, 9], 1))
